### PR TITLE
Explorer: add trace for unformed variable resolution

### PR DESCRIPTION
**Pull request by prabhatexit0**
Adds trace information for unformed variables resolution.

### CODE REVIEW

File: explorer/interpreter/resolve_unformed.cpp [hunk 1]
```diff
@@ -52,39 +68,52 @@ auto FlowFacts::TakeAction(Nonnull<const AstNode*> node, ActionType action,
...
+  });
 }
 
-static auto ResolveUnformedImpl(Nonnull<const Expression*> expression,
+static auto ResolveUnformedImpl(Nonnull<TraceStream*> trace_stream,
+                                Nonnull<const Expression*> expression,
                                 FlowFacts& flow_facts,
                                 FlowFacts::ActionType action)
     -> ErrorOr<Success> {
+  if (trace_stream->is_enabled()) {
```
Comment (zygoloid) [inline explorer/interpreter/resolve_unformed.cpp]: I worry that this is going to produce a lot of output; is it reasonable to do the same thing we did for name resolution and only log statements and declarations?

**Comment by zygoloid** [comment]
> TODO: Add detailed description.

Can you clean up the initial comment on this PR? It'll end up being used as the commit description in the new commit-queue workflow.

**Comment by prabhatexit0** [comment]
> > TODO: Add detailed description.
> 
> Can you clean up the initial comment on this PR? It'll end up being used as the commit description in the new commit-queue workflow.

Yes, I'll do that

File: explorer/interpreter/resolve_unformed.cpp (hunk 1 above)
Comment (prabhatexit0) [reply explorer/interpreter/resolve_unformed.cpp]: Yes, I think that sounds reasonable for this as well. I'll do the changes in the next commit.

**Review by zygoloid — APPROVED** [review]

**Pull request merged**